docs/decisions/0003-router-configurations-storage.rst

```diff
@@ -0,0 +1,46 @@
+1. Router Configurations Storage
+================================
+
+Status
+------
+
+Pending
+
+Context
+-------
+
+The router will use some sort of configuration to decide what events are
+to be sent to what consumers and how. These configurations can be
+defined for each partner/enterprise so we could end up having 20-50
+configurations.
+
+We have these two options for storing these configurations:
+
+-  Store configurations in Django settings (YAML configuration files).
+
+-  Store configurations in some Django model (database).
+
+Decision
+--------
+
+Keeping in mind a large number of enterprise clients, we will store
+router configurations in `Django configuration models
+<https://github.com/edx/django-config-models>`__ to keep things simple.
+
+Consequences
+------------
+
+Storing configurations in `Django configuration
+models <https://github.com/edx/django-config-models>`__ will keep the
+settings files simple and will avoid unnecessary complexities.
+
+Django configuration models have built-in caching support which would
+help us address performance-related challenges.
+
```

Comment:
Another reason to add here is that enterprise customer management might be done by teams other than the operators and putting it in the database here allows us to let customer support teams manage the settings or potentially make it self service in the future.

Reply:
Updated.